version: '3.5'

services:
  redis:
    container_name: redis
    image: redis

  ui:
    container_name: ui
    image: borysekondrej/bakalarka-frontend
    env_file:
      - frontend/.env

  flask:
    container_name: flask
    image: borysekondrej/bakalarka-backend
    depends_on:
      - redis
    volumes:
      - db:/app/bakalarka3/db
      - flask-logs:/app/bakalarka3/log
    env_file:
      - backend/.env

  sslyze_scanner_new:
    container_name: sslyze_scanner_new
    image: borysekondrej/bakalarka-backend
    depends_on:
      - redis
      - flask
    env_file:
      - backend/.env
    entrypoint: [ "rq", "worker", "sslyze-tasks", "--url", "redis://redis"]

  # sslyze_scanner_recurring:
  #   container_name: sslyze_scanner_recurring
  #   image: borysekondrej/bakalarka-backend
  #   depends_on:
  #     - redis
  #     - flask
  #   env_file:
  #     - sslyze_rq/.env

  nginx:
    container_name: nginx
    build: nginx
    depends_on:
      - ui
      - flask
    ports:
      - "443:443"
    # network_mode: "host" # this was workaround for https://github.com/gliderlabs/docker-alpine/issues/307
    # this however then breaks connection to flask:5000

volumes:
  db:
  flask-logs:

# The following was test to fix the problem, when nginx didn't have access to internet for apk update in build.    
# It was reproducible on VM Ubuntu 18.04.
# networks:                                
#   default:                               
#     driver: bridge
#     driver_opts:                         
#       com.docker.network.driver.mtu: 1400